src/websocket/notification.gateway.ts

```diff
@@ -1,35 +1,60 @@
-import { Injectable } from '@nestjs/common';
-import {
-  ConnectedSocket,
-  SubscribeMessage,
-  WebSocketGateway,
-  WebSocketServer,
-} from '@nestjs/websockets';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { WebSocketGateway, WebSocketServer } from '@nestjs/websockets';
 import { Server, Socket } from 'socket.io';
+import { AuthService } from '../auth/auth.service';
+import { UserRepository } from '../user/user.repository';
 
 @Injectable()
 @WebSocketGateway()
 export class NotificationGateway {
+  constructor(
+    private readonly authService: AuthService,
+    private readonly userRepository: UserRepository,
+  ) {}
+
   @WebSocketServer()
   server: Server;
 
-  getNotificationRoomName(): string {
-    return `recipe-notification-room`;
+  getRoomWithId(userId: number): string {
+    return `notification-userId-${userId}`;
+  }
+
+  async handleConnection(socket: Socket) {
+    const token: string =
+      socket.handshake.headers['authorization'].split(' ')[1];
```

Comment:
```suggestion
      socket.handshake.headers.authorization;
```
it should also work, you can try it